packages/run-protocol/src/vaultFactory/liquidateIncrementally.js

```diff
@@ -0,0 +1,322 @@
+/* eslint-disable no-await-in-loop */
+// no @ts-check
+
+import { E } from '@endo/eventual-send';
+import {
+  getAmountOut,
+  makeRatio,
+  assertProposalShape,
+  offerTo,
+  natSafeMath as NatMath,
+} from '@agoric/zoe/src/contractSupport/index.js';
+import { AmountMath } from '@agoric/ertp';
+import { Far } from '@endo/marshal';
+
+import { ceilMultiplyBy } from '@agoric/zoe/src/contractSupport/ratio';
+import { makeTracer } from '../makeTracer.js';
+// import { quote } from '@agoric/assert';
+
+const { details: X } = assert;
+const trace = makeTracer('LiqI');
+
+/**
+ * This contract liquidates a payment in multiple tranches to limit the
+ * price impact an the AMM of any one sale. Each block it will compute
+ * a tranche of collateral to sell, where the size is a function of
+ * the amount of that collateral in the AMM pool and the desired price impact.
+ * It then gets 3 prices for the current tranche:
+ * - AMM quote - compute xYK locally based on the pool sizes
+ * - Reserve quote - based on a low price at which the Reserve will purchase
+ * - Oracle quote - provide by the oracle that triggered liquidation
+ * - Oracle limit - a limit that is DIVERGENCE_TOLERANCE below the
+ *   Oracle quote.
+ * These sell based on the following:
+ * - If the AMM quote is *below* the oracle limit, skip this block
+ *   to wait for oracle and AMM prices to converge.
+ *    - should it still sell to teh Reserve here?
```

Comment:
likely to be missed later. better to put with the `TODO integrate the reserve`

Reply:
I expanded the TODO.